### ISSUE TITLE
Paypal Order creation fails due to wrong decimal precision

### ISSUE THREAD

**Issue by LucaGallinari** [Sylius/PayPalPlugin#183]
Sometimes calls to https://api.paypal.com/v2/checkout/orders return a 422 status code response with the following content:
```
Array(
    [name] => UNPROCESSABLE_ENTITY
    [details] => Array(
            [0] => Array(
                    [field] => /purchase_units/@reference_id=='XXX'/amount/value
                    [value] => 242.79999999999998
                    [issue] => DECIMAL_PRECISION
                    [description] => If the currency supports decimals, only two decimal place precision is supported.
                )
        )
    [message] => The requested action could not be performed, semantically incorrect, or failed business validation.
    [debug_id] => XXX
    [links] => Array(
            [0] => Array(
                    [href] => https://developer.paypal.com/docs/api/orders/v2/#error-DECIMAL_PRECISION
                    [rel] => information_link
                    [method] => GET
                )
        )
)
```

This is due to the fact that decimal precision is not properly enforced before calling the Paypal Create Order API: it should have exactly 2 decimals. Prices/amounts stored in the DB could be divided by 100 and you should always obtain a number with 2 decimals but the PayPalItemDataProvider does some sums and multiplications on the amounts and it lead to a floating point error.

This could be solved in CreateOrderApi class by enforcing decimal for all amount while setting up data, with something like this:
`number_format($amount, 2, '.', '')`

---------------

Moreover, but maybe it could be discussed in a separate issue, Paypal seems to require a different decimal precision based on the currency: https://developer.paypal.com/docs/api/reference/currency-codes/

**Comment by jeandanyel** [comment]
#190

**Comment by LucaGallinari** [comment]
Fixed by a375013
